### ISSUE TITLE
Enhancement: only expose the path in `ValidationException`s

### ISSUE THREAD

**Issue by floxay** [litestar-org/litestar#3061]
### Summary

Currently `ValidationException` exposes the full URL in the error response, leaking internal IP(s) or other similar infra related information.

Relevant code:
https://github.com/litestar-org/litestar/blob/d52caee70c6fa6c6806f7a80833b872118b3fc89/litestar/_signature/model.py#L122 https://github.com/litestar-org/litestar/blob/d52caee70c6fa6c6806f7a80833b872118b3fc89/litestar/_kwargs/extractors.py#L110  `{connection.url}` -> `{connection.url.path}`

### Basic Example

```json
{
    "status_code": 400,
    "detail": "Missing required query parameter 'q' for url http://127.0.0.1:8000/"
}
```
and
```json
{
    "status_code": 400,
    "detail": "Validation failed for GET http://127.0.0.1:8000/?q=s",
    "extra": ...
}
```
would become
```json
{
    "status_code": 400,
    "detail": "Missing required query parameter 'q' for path /"
}
```
```json
{
    "status_code": 400,
    "detail": "Validation failed for GET /",
    "extra": ...
}
```

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3061">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3061/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/3061/pledge.svg">
</picture>
</a>

**Comment by provinzkraut** [comment]
Sounds good to me. Do you want to open a PR for this?

**Comment by floxay** [comment]
I have just realized that by simply doing `url.path` the query parameters will be gone as well.

So, for example;
`Validation failed for GET http://127.0.0.1:8000/?q=s`
would become
`Validation failed for GET /`
instead of the originally mentioned
`Validation failed for GET /?q=s`

It makes sense for the 'feature request' as it really is just the path that's there but my example was incorrect. (I have now edited it in the original post)

**Comment by floxay** [comment]
Shouldn't have this auto-closed when #3064 got merged? Or is it still open on purpose?

**Comment by provinzkraut** [comment]
The PR was merged into `develop`. GitHub does not consider this merged as it's not the default branch. We do have a workflow that's supposed to close these issues, but it currently has an issue when the PR originates from a fork (see the [failing workflow here](https://github.com/litestar-org/litestar/actions/runs/7902411405)).
